File: apps/web/app/(protected)/documents/feature/types/translate.ts
// -------------------------
// Frontend
// -------------------------
export type TranslateSentence = {
  sentId: number;
  source: string;
  target: string;
};

export type TranslateParagraph = {
  parId: number;
  sentences: TranslateSentence[];
};

export type TranslateResponse = {
  sourceText: string;
  targetText: string;
  paragraphs: TranslateParagraph[];
};

// -------------------------
// Backend
// -------------------------
export type TranslateSentenceDTO = {
  sent_id: number;
  source: string;
  target: string;
};

export type TranslateParagraphDTO = {
  par_id: number;
  sentences: TranslateSentenceDTO[];
};

export type TranslateResponseDTO = {
  source_text: string;
  target_text: string;
  paragraphs: TranslateParagraphDTO[];
};

File: packages/babeling-nlp/src/babeling_nlp/schemas/translate.py
from pydantic import BaseModel


class InputSentence(BaseModel):
    sent_id: int
    source: str

class InputParagraph(BaseModel):
    par_id: int
    sentences: list[InputSentence]

class TranslationRequestPayload(BaseModel):
    paragraphs: list[InputParagraph]

class OutputSentence(BaseModel):
    sent_id: int
    target: str

class OutputParagraph(BaseModel):
    par_id: int
    sentences: list[OutputSentence]

class TranslationResponsePayload(BaseModel):
    paragraphs: list[OutputParagraph]